backend/src/users/users.repository.ts

```diff
@@ -0,0 +1,53 @@
+import { Injectable } from '@nestjs/common';
+import { CreateUserDto } from './dto/create-user.dto';
+import { UpdateUserDto } from './dto/update-user.dto';
+import { PrismaService } from 'src/prisma/prisma.service';
+
+@Injectable()
+export class UsersRepository {
+  constructor(private readonly prisma: PrismaService) {}
+
+  async createSession(userId: number, token: string) {
+    return await this.prisma.session.create({
+      data: {
+        userId,
+        token,
+      },
+    });
+  }
+
+  async create(createUserDto: CreateUserDto) {
+    return await this.prisma.user.create({
+      data: createUserDto,
+    });
+  }
+
+  async findAll() {
+    return await this.prisma.user.findMany();
+  }
+
+  async findLogin(login: string) {
+    return await this.prisma.user.findFirst({
+      where: { login },
+    });
+  }
+
+  async findOne(id: number) {
+    return await this.prisma.session.findFirst({
+      where: { userId: id },
+    });
+  }
+
+  async update(id: number, updateUserDto: UpdateUserDto) {
+    return await this.prisma.user.update({
```

Comment:
Vale pra todos também, uma função `async` já retorna uma promise, então não precisa do `await`
```suggestion
    return this.prisma.user.update({
```